pkg/hooks/manager.go

```diff
@@ -95,46 +98,83 @@ func (m *hookManager) Collect(numberOfRequestedHookSidecars uint, timeout time.D
 	return nil
 }
 
-// TODO: Handle sockets in parallel, when a socket appears, run a goroutine trying to read Info from it
 func (m *hookManager) collectSideCarSockets(numberOfRequestedHookSidecars uint, timeout time.Duration) (map[string][]*callBackClient, error) {
 	callbacksPerHookPoint := make(map[string][]*callBackClient)
 	processedSockets := make(map[string]bool)
 
-	timeoutCh := time.After(timeout)
+	if numberOfRequestedHookSidecars == 0 {
+		return callbacksPerHookPoint, nil
+	}
+
+	var mutex sync.Mutex
+	errorCh := make(chan error, numberOfRequestedHookSidecars)
 
-	for uint(len(processedSockets)) < numberOfRequestedHookSidecars {
+	err := wait.PollImmediately(time.Second, timeout, func(_ context.Context) (bool, error) {
 		sockets, err := os.ReadDir(m.hookSocketSharedDirectory)
 		if err != nil {
-			return nil, err
+			return false, err
 		}
 
+		var wg sync.WaitGroup
 		for _, socket := range sockets {
-			select {
-			case <-timeoutCh:
-				return nil, fmt.Errorf("Failed to collect all expected sidecar hook sockets within given timeout")
-			default:
-				if _, processed := processedSockets[socket.Name()]; processed {
-					continue
-				}
-
-				callBackClient, notReady, err := processSideCarSocket(filepath.Join(m.hookSocketSharedDirectory, socket.Name()))
-				if notReady {
+			mutex.Lock()
+			if _, processed := processedSockets[socket.Name()]; processed {
+				mutex.Unlock()
+				continue
+			}
+			processedSockets[socket.Name()] = true
+			mutex.Unlock()
+
+			wg.Add(1)
+			go func(socketFile os.DirEntry) {
+				defer wg.Done()
+
+				callBackClient, notReady, err := processSideCarSocket(filepath.Join(m.hookSocketSharedDirectory, socketFile.Name()))
+				if err != nil {
+					if !notReady {
```

Comment:
I think this is wrong!
Currently when `notReady` is true, err is nil.
I think it should be:
```golang
				if notReady {
					log.Log.Info("Sidecar server might not be ready yet, retrying in the next iteration")
					mutex.Lock()
					delete(processedSockets, socketFile.Name())
					mutex.Unlock()
					return
				}
				if err != nil {
					select {
					case errorCh <- fmt.Errorf("failed to process sidecar socket %s: %v", socketFile.Name(), err):
					default:
						log.Log.Errorf("Error channel full, dropping error: %v", err)
					}
					return
				}
```

Reply:
You're absolutely correct, this was an oversight on my part due to how `notReady` has the connotation of an error. Thanks!